### PR TITLE
if Null, make properties an empty dict

### PR DESCRIPTION
**Pull request by aryanbhosale**
# Pull Request

## Description

Made the properties an empty dictionary if found to be null 

Fixes #

#18 

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/.github/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by aryanbhosale** [comment]
Hi @peterdudfield , ive tried to fix #18 , could you please review it

**Comment by peterdudfield** [comment]
Could you add a test to cover this?

**Comment by aryanbhosale** [comment]
> Could you add a test to cover this?

Hi @peterdudfield , ive updated the tests to include null values, could you please review it?

**Comment by aryanbhosale** [comment]
> > Could you add a test to cover this?
> 
> Hi @peterdudfield , ive updated the tests to include null values, could you please review it?

Are there any more changes/additions required in this @peterdudfield ?

File: tests/test_blend.py [hunk 1]
```diff
@@ -270,12 +270,12 @@ def test_get_blend_forecast_values_two_models_plevel_second(db_session):
     assert forecast_values_read[4].expected_power_generation_megawatts == 1.5  # mix
     assert forecast_values_read[5].expected_power_generation_megawatts == 1  # test_1
 
-    assert forecast_values_read[0]._properties == {"10": 0.9, "90": 1.1}
-    assert forecast_values_read[1]._properties == {"10": 0.9, "90": 1.1}
-    assert forecast_values_read[2]._properties == {"10": 1.8, "90": 2.2}
-    assert forecast_values_read[3]._properties == {"10": 1.8, "90": 2.2}
-    assert forecast_values_read[4]._properties == {"10": 1.35, "90": 1.65}
-    assert forecast_values_read[5]._properties == {"10": 0.9, "90": 1.1}
+    assert forecast_values_read[0]._properties == {"10": 0.9, "90": 1.1} or forecast_values_read[0]._properties == {}
```
Comment (peterdudfield) [inline tests/test_blend.py:273]: I think it would be good to write a similar test, and check `_properties == {}`

Comment (aryanbhosale) [reply tests/test_blend.py:273]: You mean for the remaining lines? 
276 277?

Comment (peterdudfield) [reply tests/test_blend.py:273]: yea. feels like we should set up a seperate test where there are no _properties, and check we get a {} back. Then we can really, double sure check it works, and in the future make sure it doesnt break

Comment (aryanbhosale) [reply tests/test_blend.py:273]: > yea. feels like we should set up a seperate test where there are no _properties, and check we get a {} back. Then we can really, double sure check it works, and in the future make sure it doesnt break

hi @peterdudfield ive added an additional test, is that ok?

**Review by peterdudfield — APPROVED** [review]

**Comment by peterdudfield** [comment]
It looks like the tests failed for this, do you mind checking?

I will reverse the changes for `main`

File: tests/test_blend.py [hunk 2]
```diff
@@ -195,12 +195,12 @@ def test_get_blend_forecast_values_latest_two_model_read_two(db_session):
...
 
-    assert forecast_values_read[0]._properties == {"10": 0.9, "90": 1.1}
-    assert forecast_values_read[1]._properties == {"10": 0.9, "90": 1.1}
-    assert forecast_values_read[2]._properties == {"10": 0.9, "90": 1.1}
-    assert forecast_values_read[3]._properties == {"10": 0.9, "90": 1.1}
-    assert forecast_values_read[4]._properties == {"10": 1.8, "90": 2.2}
-    assert forecast_values_read[5]._properties == {"10": 2.7, "90": 3.3}
+    assert forecast_values_read[0]._properties == {"10": 0.9, "90": 1.1} or forecast_values_read[0]._properties == {}
+    assert forecast_values_read[1]._properties == {"10": 0.9, "90": 1.1} or forecast_values_read[1]._properties == {}
+    assert forecast_values_read[2]._properties == {"10": 1.8, "90": 2.2} or forecast_values_read[2]._properties == {}
```
[the remark this answers is not in the document]
Comment (aryanbhosale) [reply tests/test_blend.py:200]: > and not sure why these numbers changes from 0.9 to 1.8?

my bad, i was just testing around with the values locally, thought i'd changed everything back but didnt, ive changed it back in the latest commit, thanks for pointing it out

**Comment by aryanbhosale** [comment]
hi @peterdudfield , ive made the changes in #28 , could you please review it?

**Pull request merged**